src/main/java/com/eternalcode/parcellockers/database/ParcelDatabaseService.java

```diff
@@ -0,0 +1,324 @@
+                                    "`priority`, " +
+                                    "`receiver`, " +
+                                    "`size`, " +
+                                    "`entryLocker`, " +
+                                    "`destinationLocker`, " +
+                                    "`sender`) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)")
+            ) {
+                statement.setString(1, parcel.uuid().toString());
+                statement.setString(2, parcel.name());
+                statement.setString(3, parcel.description());
+                statement.setBoolean(4, parcel.priority());
+                statement.setString(5, parcel.receiver().toString());
+                statement.setString(6, parcel.size().name());
+                statement.setString(7, parcel.entryLocker().toString());
+                statement.setString(8, parcel.destinationLocker().toString());
+                statement.setString(9, parcel.sender().toString());
+                statement.execute();
+                this.cache.put(parcel.uuid(), parcel);
+            }
+            catch (SQLException e) {
+                Sentry.captureException(e);
+                throw new ParcelLockersException(e);
+            }
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public CompletableFuture<Void> update(Parcel oldParcel, Parcel newParcel) {
+        return CompletableFuture.runAsync(() -> {
+            try (Connection connection = this.dataSource.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(
+                         "UPDATE `parcels` SET " +
+                                 "`name` = ?, " +
+                                 "`description` = ?, " +
+                                 "`priority` = ?, " +
+                                 "`receiver` = ?, " +
+                                 "`size` = ?, " +
+                                 "`entryLocker` = ?, " +
+                                 "`destinationLocker` = ?, " +
+                                 "`sender` = ? " +
+                                 "WHERE `uuid` = ?"
+                 )
+            ) {
+                statement.setString(1, newParcel.name());
+                statement.setString(2, newParcel.description());
+                statement.setBoolean(3, newParcel.priority());
+                statement.setString(4, newParcel.receiver().toString());
+                statement.setString(5, newParcel.size().name());
+                statement.setString(6, newParcel.entryLocker().toString());
+                statement.setString(7, newParcel.destinationLocker().toString());
+                statement.setString(8, newParcel.sender().toString());
+                statement.setString(9, oldParcel.uuid().toString());
+                statement.execute();
+                this.cache.remove(oldParcel.uuid());
+                this.cache.put(newParcel.uuid(), newParcel);
```

Comment:
zdaje mi się, że jak `uuid` się nie zmieni (czyli klucz mapki) to wartość do tego klucza się ustawi inna. Chodzi mi o to, że jak masz mapkę np. `Map<Integer, String> map` i masz w niej takie wartości: 
1, "dupa"
2, "xd"

to jak zrobisz `map.put(1, "lol")` to nie będzie w mapie wartości 1, dupa i 1, lol tylko będzie sam 1, lol - wartość w mapce się nadpisuję jeśli klucz się nie zmienia, więc możesz wywalić z argumentu metody oldParcel i wywalić linijkę `this.cache.remove(oldParcel.uuid());`

Reply:
mam nadzieję, że zrozumiałeś moją rozprawkę po to, aby usunąć jedną linijkę xD